### PR TITLE
feat(data-table): split data-table into atomic reusable components + theming

### PR DESCRIPTION
**Pull request by emoralesb05**
## Description

There is a need to use data-table in many different situations, so besides providing a `<data-table>` component that does some of the things, its better to have atomic components so the consumer can create their custom data-table with them (atoms to build a molecule).
### What's included?
- Atomic components of the data-table to create own encapsulating their css classes and inputs.

Usage:

``` html
     <table td-data-table>
        <th td-data-table-column>
          column header
        </th>
        <th td-data-table-column>
          column header 2
        </th>
        <tr td-data-table-row>
          <td td-data-table-cell>
            <span class="md-body-1">cell</span>
          </td>
          <td td-data-table-cell>
            <span class="md-body-1">cell 2</span>
          </td>
        </tr>
      </table>
```
- `<td-data-table>` still works, but most of the things were abstracted from it and there are components created from those abstractions (`td-search-box`, `td-paging-bar` and `TdDataTableService` (`TdDataTableService` contains the local sorting/paging/searching logic).
- [(ngModel)] support for `<td-data-table>`
- Theming for `data-table` module included with `@include covalent-theme($theme)`.
- Removed `lodash` as dependency.
- Major cleanup of css and ts code for the `data-table` module.
- Added `rowSelect` event when clicking on a row in `<td-data-table>`
#### Test Steps
- [x] `ng serve`
- [x] Go to http://localhost:4200/#/components/data-table
##### Screenshots or link to CodePen/Plunker/JSfiddle

image
##### Screenshots with paging-bar + search-box + tdDataTableService

image

### CODE REVIEW

**Comment by emoralesb05** [comment]
- [ ] Renamed css classes to use `td-` prefix since theming is applied globally.

**Comment by emoralesb05** [comment]
- [ ] Made sort icon `position: absolute` so the headers arent pushed when the `data-table` is sortable.

**Comment by asaph26** [comment]
@emoralesb05 What are your thoughts on exporting the interfaces that are used while setting up and accessing the data-table(ITdDataTableSelectEvent & ITdDataTableColumn) so that people don't need to access the specific component module to access them?

**Comment by emoralesb05** [comment]
@asaph26 Not sure i follow. Can you elaborate with an example?

**Comment by asaph26** [comment]
I was mistaken, the interfaces are actually exported. 

Was confused by the following [diff](https://github.com/Teradata/covalent/pull/128/files#diff-ab9420422cb4d1f5ad852533c7f0530bR53), columns is currently defined as type any[] instead of as ITdDataTableColumn[].

**Comment by emoralesb05** [comment]
Oh, yeah. Ill change that later on haha
That was just me being lazy and not importing the interfaces.

**Comment by emoralesb05** [comment]
- [x] Updated data table docs

**Comment by kyleledbetter** [comment]
Might be nice to have docs around custom columns like you did here:

```
      <table td-data-table>
        <th td-data-table-column>
          <md-icon>comment</md-icon>
          <span>Comments</span>
        </th>
        <th td-data-table-column 
            *ngFor="let column of columns"
            [numeric]="column.numeric">
          {{column.label}}
        </th>
        <tr td-data-table-row *ngFor="let row of basicData">
          <td td-data-table-cell (click)="openPrompt(row, 'comments')">
            <button md-button [class.md-accent]="!row['comments']">{{row['comments'] || 'Add Comment'}}</button>
          </td>
          <td td-data-table-cell *ngFor="let column of columns" [numeric]="column.numeric">
            {{row[column.name]}}
          </td>
        </tr>
      </table>
```

**Comment by kyleledbetter** [comment]
and also docs explaining how to use the other paging/search components with the datatable

```
<div layout="row" layout-align="start center" class="pad-left-sm pad-right-sm" [class.md-selected-title]="selectedRows.length && selectable">
      <span *ngIf="!searchBox.searchVisible" class="push-left-sm">
        <span *ngIf="(selectable && !selectedRows.length) || !selectable" class="md-title">Title here</span>
        <span *ngIf="selectedRows.length && selectable" class="md-body-1">{{selectedRows.length}} item(s) selected</span>
      </span>
      <td-search-box #searchBox backIcon="arrow_back" class="push-right-sm" placeholder="Search here" (searchDebounce)="search($event)" flex>
      </td-search-box>
      <button md-icon-button>
        <md-icon>more_vert</md-icon>
      </button>
    </div>
    <md-divider></md-divider>
    <td-data-table
      [data]="filteredData"
      [columns]="columns"
      [selectable]="selectable"
      [multiple]="multiple"
      [sortable]="true"
      [sortBy]="sortBy"
      [(ngModel)]="selectedRows"
      [sortOrder]="sortOrder"
      (sortChange)="sort($event)">
    </td-data-table>
    <td-paging-bar [pageSizes]="[5, 10, 15, 20]" [total]="filteredTotal" (change)="page($event)"></td-paging-bar>
```

File: src/platform/data-table/data-table.component.ts [hunk 1]
```diff
@@ -163,162 +175,124 @@ export class TdDataTableComponent implements OnInit {
...
+                                      new EventEmitter<ITdDataTableSortChangeEvent>();
+
+  /**
+   * rowSelect?: function
+   * Event emitted when a row is selected/deselected. [selectable] needs to be enabled.
+   * Emits an [ITdDataTableSelectEvent] implemented object.
+   */
+  @Output('rowSelect') onRowSelect: EventEmitter<ITdDataTableSelectEvent> = new EventEmitter<ITdDataTableSelectEvent>();
+
+  clearModel(): void {
```
Comment (mrjayhawk) [inline src/platform/data-table/data-table.component.ts]: maybe add a comment to what this method is doing?

Comment (mrjayhawk) [inline src/platform/data-table/data-table.component.ts]: actually this file has bunches of methods in it that probably need docs

Comment (emoralesb05) [inline src/platform/data-table/data-table.component.ts]: Yeah, ill add more comments around the code to make it easier to review :smile:

**Comment by sangnt1004** [comment]
I have an issue. The order of sorting is not changed on column with string on below.
31-Re
31-Op
31-Co

**Pull request merged**